mlflow/pyfunc/__init__.py

```diff
@@ -206,6 +206,100 @@
 - If you have already collected all of your model data in a single location, the second
   workflow allows it to be saved in MLflow format directly, without enumerating constituent
   artifacts.
+
+*******************************
+python_function vs. PythonModel
+*******************************
+
+When creating custom Pyfunc models, you can choose between two different interfaces:
+``python_function`` and ``PythonModel``. In short, a ``python_function`` is simply a
+python function whereas a ``PythonModel`` is a class that supports several required and optional
+methods. If your use case is simple and fits within a single predict function,
+``python_function`` is the way to go.  If you need customized data processing, custom serialization,
+or to override additional methods, you should use ``PythonModel``.
+
+Before looking at code examples, it's important to note that both methods are serialized via
+`cloudpickle <https://github.com/cloudpipe/cloudpickle>`_. cloudpickle can serialize Python
+functions, lambda functions, and locally defined classes and functions inside other functions. This
+makes cloudpickle especially useful for parallel and distributed computing where code objects need
+to be sent over network to execute on remote workers, which is a common deployment paradigm for
+MLflow.
+
+That said, cloudpickle has some limitations.
+
+- **Environment Dependency**: cloudpickle does not capture the full execution environment, so in
+    MLflow we must pass ``pip_requirements``, ``extra_pip_requirements``, or an ``input_example``,
+    the latter of which is used to infer environment dependencies.
+
+- **Object Support**: cloudpickle does not serialize objects outside of the Python data model.
+    Some relevant examples include raw files and database connections. If your program depends on
+    these, be sure to log ways to reference these objects along with your model.
+
+python_function
+###############
+If you're looking to serialize a simple python function without additional dependent methods, you
+can simply log a predict method via the keyword argument ``python_model``.
+
+.. code-block:: python
+    :caption: python_function Example
+
+    import mlflow
+    import pandas as pd
+
+    # Define a simple function to log
+    def predict(model_input):
+        return model_input.apply(lambda x: x * 2)
+
+    # Save the function as a model
+    with mlflow.start_run():
+        mlflow.pyfunc.log_model("model", python_model=predict, pip_requirements=["pandas"])
+        run_id = mlflow.active_run().info.run_id
+
+    # Load the model from the tracking server and perform inference
+    model = mlflow.pyfunc.load_model(f"runs:/{run_id}/model")
+    x_new = pd.Series([1,2,3])
+
+    print(f"Prediction:\n\t{model.predict(x_new)}")
```

Comment:
```suggestion
    prediction = model.predict(x_new)
    print(prediction)
```

Another weird sphinx thing but it treats `\n` in code as a real new line🙄
![Screenshot 2024-03-09 at 0 00 54](https://github.com/mlflow/mlflow/assets/31463517/f5b6b6d7-1890-4483-b7de-0914f1791831)